onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2893,17 +2893,25 @@ def aten_kthvalue(
     raise NotImplementedError()
 
 
+@torch_op("aten::layer_norm", trace_only=True)
 def aten_layer_norm(
-    input: TensorType,
+    input: TReal,
     normalized_shape: Sequence[int],
-    weight: Optional[TensorType] = None,
-    bias: Optional[TensorType] = None,
+    weight: Optional[TReal] = None,
+    bias: Optional[TReal] = None,
     eps: float = 1e-05,
     cudnn_enable: bool = True,
 ) -> TensorType:
     """layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True) -> Tensor"""
 
-    raise NotImplementedError()
+    axes = [-i for i in range(len(normalized_shape), 0, -1)]
+    if weight is None:
+        weight = op.CastLike(1, input)
+    if bias is None:
+        bias = op.CastLike(0, input)
+
+    result, _, _ =  _aten_native_layer_norm_onnx(input, weight, bias, axes=axes, eps=eps)
```

Comment:
I wonder if we should implement an function only for layer_norm, so we don’t have unused outputs in the onnx graph and also make it clear to users that the op is a different op. 

Also since we have opset 18 now I wonder if some None checking can be done with optional get element? For this I do want to check if the resulting function works well on ORT first